src-electron/generator/matter/app/zap-templates/templates/chip/helper.js

```diff
@@ -745,6 +745,97 @@ async function chip_access_elements(options) {
   return templateUtil.templatePromise(this.global, p);
 }
 
+/**
+ * If helper that checks if an attribute is basic or not based on language
+ * being generated for.
+ * For eg: In java, an attribute is not basic if it is either nullable, optional,
+ * array type or struct.
+ * Note: This helper should be used within an attribute block helper.
+ * Available Options:
+ * - language: Determine basic attribute based on language.
+ * example:
+ * {{#if_basic_attribute type}}
+ * type is basic
+ * {{else}}
+ * type is not basic
+ * {{/if_basic_attribute}}
+ * @param {*} type
+ * @param {*} clusterId
+ * @param {*} options
+ * @returns Promise of content
+ */
+async function if_basic_attribute(type, clusterId, options) {
+  let hash = options.hash;
+  let language = hash && hash.language ? hash.language : null;
+  if (language == 'java') {
+    let struct = null;
+    if (this.isNullable || this.isOptional || this.isArray) {
+      return options.inverse(this);
+    } else {
+      let packageIds = await templateUtil.ensureZclPackageIds(this);
+      struct = await queryZcl.selectStructByNameAndClusterId(
+        this.global.db,
+        type,
+        clusterId,
+        packageIds
+      );
+      if (struct) {
+        return options.inverse(this);
+      } else {
+        return options.fn(this);
+      }
+    }
+  } else {
+    return options.fn(this);
+  }
+}
+
+/**
+ * If helper that checks if an attribute is not supported based on language
+ * being generated for.
+ * For eg: In java, an attribute callback is not supported when it is a struct.
+ * However it is supported if it is an array of structs.
+ * Available Options:
+ * - language: attribute callback support based on language
+ * @param {*} type
+ * @param {*} isArray
+ * @param {*} clusterId
+ * @param {*} options
+ * @returns Promise of content
+ */
+async function if_unsupported_attribute_callback(
```

Comment:
Again, this is Matter-java-specific, right?  Please place this in the relevant helper file, remove the language bit.

It also feels like we should be able to build this out of existing "is this an array?" and "is this a struct type?" pieces, but since I am hoping this will just go away sometime I don't care too much about that part...

Reply:
Moved to java helper.js and language option has been removed